### PR TITLE
feat: add GTFS Loader

### LINKED ISSUE #145: Implement GTFS loader
* load GTFS from .zip file
* use gtfs-kit to extract features needed by gtfs2vec
* if possible, then don't aggregate to regions

### PR DESCRIPTION
**Pull request by piotrgramacki**
closes #145

### CODE REVIEW

**Review by RaczeQ — CHANGES_REQUESTED** [review]

File: srai/loaders/gtfs_loader.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            crs=WGS84_CRS,
+        )
+
+        result_gdf = result_gdf.merge(directions_df, how="left", on="stop_id")
+
+        result_gdf.index.name = None
+
+        return result_gdf
+
+    # FIXME: how to type this if gtfs_kit is optional?
```
[the remark this answers is not in the document]
Comment (piotrgramacki) [reply srai/loaders/gtfs_loader.py:84]: That's what I was looking for ;) Thanks!

File: srai/loaders/gtfs_loader.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        gtfs_file: Path,
+        fail_on_validation_errors: bool = True,
+        skip_validation: bool = False,
+    ) -> gpd.GeoDataFrame:
+        """
+        Load GTFS feed and calculate time aggregations for stops.
+
+        Args:
+            gtfs_file (Path): Path to the GTFS feed.
+            fail_on_validation_errors (bool): Fail if GTFS feed is invalid.
```
Comment (simonusher) [inline srai/loaders/gtfs_loader.py:53]: Maybe make it explicit here, that this is ignored if skip_validation is True?

Comment (piotrgramacki) [reply srai/loaders/gtfs_loader.py:53]: Fair point, I will add it then

File: examples/loaders/gtfs_loader.ipynb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    "2. https://transitfeeds.com/p/mpk-wroc-aw/663"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! wget -O files/example.zip https://transitfeeds.com/p/mpk-wroc-aw/663/20221221/download"
```
Comment (RaczeQ) [inline examples/loaders/gtfs_loader.ipynb:45]: We should replace this `wget` dependency and implement a dedicated download function.
https://gist.github.com/yanqd0/c13ed29e29432e3cf3e7c38467f42f51
https://stackoverflow.com/questions/37573483/progress-bar-while-download-file-over-http-with-requests

**Review by RaczeQ — CHANGES_REQUESTED** [review]

**Comment by RaczeQ** [comment]
Please add tests for new dependencies group in this file: https://github.com/srai-lab/srai/blob/main/tests/miscellaneous/test_optional_dependencies.py 
https://github.com/srai-lab/srai/blob/4d7f2b9cba03275d42d4447fb82f8b0c7d08f20b/tests/miscellaneous/test_optional_dependencies.py#L37-L44

File: examples/loaders/gtfs_loader.ipynb (hunk 3 above)
Comment (piotrgramacki) [reply examples/loaders/gtfs_loader.ipynb]: Done ;)

File: pyproject.toml [hunk 4]
```diff
@@ -122,6 +128,10 @@ no_implicit_optional = true
 check_untyped_defs = true
 warn_return_any = true
 
+[[tool.mypy.overrides]]
+module = ["requests.*"]
+ignore_missing_imports = true
```
[the remark this answers is not in the document]
Comment (piotrgramacki) [reply pyproject.toml]: I forgot about it. I changed it in the same way as you in #167

**Review by simonusher — APPROVED** [review]

File: examples/loaders/utils.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+"""Utility functions for loaders examples."""
+import requests
+from tqdm import tqdm
+
+
+def download(url: str, fname: str, chunk_size: int = 1024) -> None:
```
Comment (RaczeQ) [inline examples/loaders/utils.py:6]: `download_file` and `filename` would be more descriptive
```suggestion
def download_file(url: str, filename: str, chunk_size: int = 1024) -> None:
```

**Review by RaczeQ — APPROVED** [review]

**Review by Calychas — COMMENTED** [review]
Post factum LGTM

**Pull request merged**